syntax = "proto3";

package transaction_verification;

service TransactionVerificationService {
    rpc InitOrder(InitOrderRequest) returns (InitOrderResponse);
    rpc CheckBooks(EventRequest) returns (EventResponse);
    rpc CheckUserFields(EventRequest) returns (EventResponse);
    rpc CheckCardFormat(EventRequest) returns (EventResponse);
    rpc ClearOrder(ClearOrderRequest) returns (ClearOrderResponse);
}

message InitOrderRequest {
    string order_id = 1;
    map<string, string> user_data = 2;
    repeated string books = 3;
    string credit_card = 4;
}

message InitOrderResponse {
    bool success = 1;
    string message = 2;
    map<string, int32> vector_clock = 3;
}

message EventRequest {
    string order_id = 1;
}

message EventResponse {
    bool is_success = 1;
    string message = 2;
    map<string, int32> vector_clock = 3;
}

message ClearOrderRequest {
    string order_id = 1;
    map<string, int32> final_vector_clock = 2;
}

message ClearOrderResponse {
    string status = 1;
}
